mobile_app/trigger_build.py

```diff
@@ -0,0 +1,129 @@
+"""
+Triggers a new app build by making a branch for it.
+"""
+
+import argparse
+import git
+import logging
+import json
+import os
+import sys
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+CONFIG_FILE = "CONFIGURATION"
+EXPECTED_ENVIRONMENT_VARIABLES = [
+    "CONFIG_REPO",
+    "CONFIG_PATH",
+    "CONFIG_REVISION",
+    "CODE_REPO",
+    "CODE_REVISION",
+]
+
+
+class MissingEnvironmentVariable(Exception):
+    """
+    Indicates that an expected environment variable was not found.
+    """
+    def __init__(self, variable):
+        self.variable = variable
+        super(MissingEnvironmentVariable, self).__init__()
+
+    def __str__(self):
+        return "Missing environment variable: {variable}".format(
+            variable=self.variable
+        )
+
+
+class BranchAlreadyExists(Exception):
+    """
+    Indicates that the branch we're trying to create already exists
+    """
+    def __init__(self, branch_name):
+        self.branch_name = branch_name
+        super(BranchAlreadyExists, self).__init__()
+
+    def __str__(self, branch_name):
+        return "Branch already exists: {branch}".format(
+            branch=branch_name
+        )
+
+
+def _build_parser():
+    """
+        Constructs a command line option parser
+
+        Returns:
+            ArgumentParser: The option parser
+    """
+    parser = argparse.ArgumentParser(
+        description="""
+        Commits a list of environment variables to trigger a new build
+        in that environment
+        """
+    )
+    parser.add_argument(
+        '--branch-name',
+        nargs='?',
+        required=True
+    )
+    parser.add_argument(
+        '--repo-path',
+        nargs='?',
+        default='.'
+    )
+    return parser
+
+
+def run_trigger_build(raw_args, environ):
```

Comment:
Nit: per Clean-Code convention, have you considered reordering the functions in this module so they read top-down?